### PR TITLE
Refactoring and adding a basic test

### PR DESCRIPTION
**Pull request by Chocrates**
Addresses Issue #7 
Adds the test infrastructure and a couple basic tests to build from. 
Does some basic refactoring, though we could probably do more as we expand this method.  

**Note:** This is going to have some nasty merge conflict with #12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
     "format-check": "prettier --check **/*.js",
     "lint": "eslint index.js",
     "package": "ncc build index.js -o dist",
-    "test": "eslint index.js && jest",
+    "test": "eslint index.js && jest --collect-coverage",
```
Comment (froi) [inline package.json]:
```suggestion
    "pretest": "npm run lint",
    "test": "jest --collect-coverage",
```

We should use npm's hooks to run the linter before the test are run. We should also reuse the lint npm script.

Comment (froi) [reply package.json]: For reference https://docs.npmjs.com/misc/scripts

File: tests/main.test.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+let functions = {
+  createInvitation: jest.fn().mockReturnValue(true),
+  getContents: jest.fn().mockReturnValue({
+    data: {
+      content: Buffer.from(JSON.stringify(parsingRules)).toString("base64")
+    }
+  }),
+
+  setFailed: jest.fn(msg => {
+    console.log(`MOCK ERROR: ${msg}`);
```
Comment (froi) [inline tests/main.test.js]:
```suggestion
    console.error(`MOCK ERROR: ${msg}`);
```

**Comment by Chocrates** [comment]
Changes accepeted, ready for another review, @froi

**Pull request merged**